cluster/watchdog.go

```diff
@@ -0,0 +1,112 @@
+package cluster
+
+import (
+	"sync"
+
+	log "github.com/Sirupsen/logrus"
+)
+
+// Watchdog listen to cluster events ans handle container rescheduling
+type Watchdog struct {
+	sync.Mutex
+	cluster Cluster
+}
+
+// Handle cluster callbacks
+func (w *Watchdog) Handle(e *Event) error {
+	switch e.Status {
+	case "engine_reconnect":
+		go w.removeDuplicateContainers(e.Engine)
+	case "engine_disconnect":
+		go w.rescheduleContainers(e.Engine)
+	case "die", "destroy", "kill", "oom", "start", "stop", "rename":
+		go w.reschedulePendingContainers()
```

Comment:
Any reason this trigger whole cluster reschedule, instead of just this engine? I think there could be a lot of events for a big cluster.